### PR TITLE
IDX-1428: common indexing

### LINKED ISSUE #1428: [common] common/indexing mod
**Summary**

1. extract indexing algo
2. search indexing algo
3. write/read api with MetaStore or DFS

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary
This patch only impl the min_max and sparse index create and make a spec for pruning with `apply_index`(always return true now).
We will impl the pruning logic in another issue https://github.com/datafuselabs/datafuse/issues/1552.

Tasks:
- [x] extract indexing from the blocks which sorted for one parquet.file
     - [x] min/max indexing
     - [x] sparse indexing

## Changelog

- New Feature

## Related Issues

Fixes #1428

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/datablocks/src/data_block.rs [hunk 1]
```diff
@@ -132,6 +117,20 @@ impl DataBlock {
         }
     }
 
+    /// Take the first data value of the column.
+    pub fn first(&self, col: &str) -> Result<DataValue> {
+        let series = self.try_array_by_name(col)?;
+        let values = series.to_values()?;
```
Comment (sundy-li) [inline common/datablocks/src/data_block.rs:123]: `to_values()` is too heavy,  try  `try_get(idx)`

File: common/datablocks/src/data_block.rs [hunk 2]
```diff
@@ -132,6 +117,20 @@ impl DataBlock {
...
+        Ok(values[0].clone())
+    }
+
+    /// Take the last data value of the column.
+    pub fn last(&self, col: &str) -> Result<DataValue> {
+        let series = self.try_array_by_name(col)?;
+        let values = series.to_values()?;
+        Ok(values[values.len() - 1].clone())
```
Comment (sundy-li) [inline common/datablocks/src/data_block.rs:131]: .

File: common/datablocks/src/data_block.rs (hunk 1 above)
Comment (sundy-li) [reply common/datablocks/src/data_block.rs:123]: There is already a method in `DataColumn` named try_get

**Review by zhang2014 — APPROVED** [review]
LGTM

File: common/datablocks/src/data_block.rs (hunk 1 above)
Comment (BohuTANG) [reply common/datablocks/src/data_block.rs:123]: Ok, I will update it in another patch.

**Pull request merged**